tool/util/shared.go

```diff
@@ -83,6 +88,28 @@ func IsCompileCommand(line string) bool {
 	return true
 }
 
+// GetOtelGoCacheEnv returns a GOCACHE environment variable
+// pointing to a temporary build cache directory to avoid polluting the user's go build environment.
+func GetOtelGoCacheEnv() ([]string, error) {
```

Comment:
It's not a common utility function. Please move it to the package where it's used.

The name GetOtelGoCacheEnv is problematic - gocacheenv is unrelated to otel. I suggest using GetTempGoCache and having it return only the path, with the key=value form assembled at the call site.